### PR TITLE
Adding an example to show N1QL queries against custom FTS index(es)

### PR DESCRIPTION
**Pull request by abhinavdangeti**
Custom FTS index => FTS index with a custom type mapping

### CODE REVIEW

**Review by amarantha-k — APPROVED** [review]

File: modules/n1ql/pages/n1ql-language-reference/searchfun.adoc [hunk 1]
```diff
@@ -300,6 +300,36 @@ This query returns 10 results, and the results are ordered, as specified by the
...
+  {
+    "id": "airline_2395"
+  },
+  {
+    "id": "airline_2657"
+  },
+...
+----
+
+If the FTS index being queried has it's default mapping disabled and has a custom type mapping defined ("airline" in the above example), the query would need to explicitly specify the "type".
```
Comment (simon-dew) [inline modules/n1ql/pages/n1ql-language-reference/searchfun.adoc:328]: Which index is this example supposed to use? None of the demonstration indexes use "airline" as a custom type mapping. If the example doesn't use one of the specified [demonstration indexes](https://docs.couchbase.com/server/6.5/fts/fts-demonstration-indexes.html), it won't be reproducible by the user.

File: modules/n1ql/pages/n1ql-language-reference/searchfun.adoc [hunk 2]
```diff
@@ -300,6 +300,36 @@ This query returns 10 results, and the results are ordered, as specified by the
 As an alternative, you could limit the number of results and order them using the N1QL xref:n1ql-language-reference/limit.adoc[LIMIT] and xref:n1ql-language-reference/orderby.adoc[ORDER BY] clauses.
 ====
 
+.Search against an FTS index that carries a custom type mapping
+====
+[source,n1ql]
+----
+SELECT META(t1).id
+FROM `travel-sample` AS t1
+WHERE t1.type = "airline" AND SEARCH(t1.country, "United States");
```
Comment (simon-dew) [inline modules/n1ql/pages/n1ql-language-reference/searchfun.adoc:309]: I'm not sure what this example is supposed to be demonstrating. Is the SEARCH function here meant to have an index hint with a custom mapping?

**Review by simon-dew — COMMENTED** [review]
sorry for late comments on this PR, I only noticed these changes when I started work on DOC-6237.

File: modules/n1ql/pages/n1ql-language-reference/searchfun.adoc (hunk 1 above)
Comment (abhinavdangeti) [reply modules/n1ql/pages/n1ql-language-reference/searchfun.adoc:328]: Hmm, there's some documentation on how to create a custom type mapping here: https://docs.couchbase.com/server/6.5/fts/fts-creating-indexes.html, but for the "type" hotel documents. Would you think it'd do better to update the type here to "hotel" and point them to the "fts-creating-pages" page on how to set them up.

File: modules/n1ql/pages/n1ql-language-reference/searchfun.adoc (hunk 2 above)
Comment (abhinavdangeti) [reply modules/n1ql/pages/n1ql-language-reference/searchfun.adoc:309]: The explicit t.type = "airline" becomes necessary when an index with a custom mapping is used.

File: modules/n1ql/pages/n1ql-language-reference/searchfun.adoc (hunk 1 above)
Comment (simon-dew) [reply modules/n1ql/pages/n1ql-language-reference/searchfun.adoc:328]: Resolved by #1105

File: modules/n1ql/pages/n1ql-language-reference/searchfun.adoc (hunk 2 above)
Comment (simon-dew) [reply modules/n1ql/pages/n1ql-language-reference/searchfun.adoc:309]: Resolved by #1105

**Pull request merged**